atlp-rwanda/cabal-bn-be
# #181339468 Setting Database and Sequelize

### What does this PR do?

Configure Bare Foot Nomad to use PostgreSQL as DB Engine and Sequelize ORM

### Description of Task to be completed?

Create Postgres database on Heroku.

Use sequelize as our ORM on the API. Set up the migrations and models structure, have a dummy user table and model setup to test the whole configuration.

### How should this be manually tested?

- After cloning this repository, install all packages on your local machine with **`yarn install`** 
- Make sure that you have Postgres installed and then create dev and test databases.
- Create a **`.env`** file in root directory of your project and copy variables provided in **`.env.sample`** file; assign **`DEV_DATABASE_URL`** with your local dev database URL and **`TEST_DATABASE_URL`** with your test db URL to test the connection. **Example:** `postgres://<db_user>:<db_password>@127.0.0.1:5432/<dev_database_name>`
- Run the project using **`yarn run dev`** script.
- To **migrate** database run : `yarn run dbmigrate`
- To seed the database run: `yarn run seeding`
Notice: If you have an existing similar table in the database you may have an error. In that case run `yarn run migrate:reset` to reset the database.
You can also do all those steps above (migrating, seeding) at the same time with one command: **`yarn run dbmigrate:all`**

After the steps above, check in your database with pgAdmin or CLI to check if you can see your seeded data.

### Relevant Pivotal Tracker Stories

Click -> [181339468](https://www.pivotaltracker.com/story/show/181339468)

### Screenshots (if appropriate)
![term](https://user-images.githubusercontent.com/60324751/156344087-fc9eef13-e1c9-4f81-a276-8bf64363466a.png)
![db](https://user-images.githubusercontent.com/60324751/156344515-470197ea-1777-44bc-8efa-0354e51ef90b.png)
